packages/codemirror/src/editor.ts

```diff
@@ -841,6 +841,42 @@ class CodeMirrorEditor implements CodeEditor.IEditor {
     return caret;
   }
 
+  /**
+   * Schedule an editor integrity check.
+   */
+  private _scheduleCheck(): void {
+    window.clearTimeout(this._checkTimer);
+    this._checkTimer = window.setTimeout(() => {
```

Comment:
This is allocating a new closure on every keystroke. You could use a `setInterval` call to set up a persistent loop, and only do work if there has been a change since the last time you checked.

Reply:
Updated